tests/mechanic/launcher_test.py

```diff
@@ -139,6 +140,25 @@ def test_daemon_start_stop(self, process, wait_for_pidfile, node_config, ms, cfg
         proc_launcher.keep_running = False
         proc_launcher.stop(nodes)
         self.assertTrue(kill.called)
+             
+    @mock.patch('os.path.join', return_value="/java_home/bin")
```

Comment:
This should not be necessary and this test should work without mocking `os.path.join`.